controllers/front/OrderConfirmationController.php

```diff
@@ -58,17 +61,21 @@ public function init()
         $this->secure_key = Tools::getValue('key', false);
         $order = new Order((int) ($this->id_order));
 
-        if (!$this->id_order || !$this->id_module || !$this->secure_key || empty($this->secure_key)) {
+        if (!$this->id_order || (!$this->isFreeOrder && !$this->id_module) || !$this->secure_key || empty($this->secure_key)) {
```

Comment:
Isn't `$this->isFreeOrder` always false when we reach this condition?

Or am I missing something?

Reply:
ah didn't see the new line in `checkFreeOrder()` method 👍